astropy/coordinates/tests/test_stokes_coord.py

```diff
@@ -0,0 +1,169 @@
+
+
+def test_undefined():
+    sk = StokesCoord(np.arange(-10, 7))
+    assert_equal(
+        sk.symbol,
+        np.array(
+            [
+                "?",
+                "?",
+                "YX",
+                "XY",
+                "YY",
+                "XX",
+                "LR",
+                "RL",
+                "LL",
+                "RR",
+                "?",
+                "I",
+                "Q",
+                "U",
+                "V",
+                "?",
+                "?",
+            ]
+        ),
+    )
+
+
+def test_undefined_init():
+    with pytest.raises(Exception, match="Unknown stokes symbols.*Spam"):
+        StokesCoord("Spam")
+
+
+def test_custom_symbol_mapping():
+    custom_mapping = {
+        10000: StokesSymbol("A"),
+        10001: StokesSymbol("B"),
+        10002: StokesSymbol("C"),
+        10003: StokesSymbol("D"),
+    }
+
+    # Check that we can supply a custom mapping
+    with custom_stokes_symbol_mapping(custom_mapping):
+        values = [0.6, 1.7, 10000.1, 10002.4]
+        sk1 = StokesCoord(values)
+        assert repr(sk1) == "StokesCoord(['I', 'Q', 'A', 'C'])" == str(sk1)
+        assert_equal(sk1.value, values)
+        assert_equal(sk1.symbol, np.array(["I", "Q", "A", "C"]))
+
+    # Check that the mapping is not active outside the context manager
+    assert_equal(sk1.symbol, np.array(["I", "Q", "?", "?"]))
+
+    # But not for new StokesCoords
+    sk2 = StokesCoord(values)
+    assert_equal(sk2.symbol, np.array(["I", "Q", "?", "?"]))
+
+
+def test_custom_symbol_mapping_overlap():
+    # Make a custom mapping that overlaps with some of the existing values
+
+    custom_mapping = {
+        3: StokesSymbol("A"),
+        4: StokesSymbol("B"),
+        5: StokesSymbol("C"),
+        6: StokesSymbol("D"),
+    }
+
+    with custom_stokes_symbol_mapping(custom_mapping):
+        sk = StokesCoord(np.arange(1, 7))
+        assert_equal(sk.symbol, np.array(["I", "Q", "A", "B", "C", "D"]))
+
+
+def test_custom_symbol_mapping_replace():
+    # Check that we can replace the mapping completely
+
+    custom_mapping = {
+        3: StokesSymbol("A"),
+        4: StokesSymbol("B"),
+        5: StokesSymbol("C"),
+        6: StokesSymbol("D"),
+    }
+
+    with custom_stokes_symbol_mapping(custom_mapping, replace=True):
+        sk = StokesCoord(np.arange(1, 7))
+        assert_equal(sk.symbol, np.array(["?", "?", "A", "B", "C", "D"]))
+
+
+def test_comparison_scalar():
+    sk = StokesCoord(np.arange(1, 6))
+    assert_equal(np.equal(sk, "Q"), [False, True, False, False, False])
+    assert_equal(np.equal("Q", sk), [False, True, False, False, False])
+    assert_equal("Q" == sk, [False, True, False, False, False])
+    assert_equal(sk == 1, [True, False, False, False, False])
+    assert_equal(sk == "Q", [False, True, False, False, False])
+    # TODO: Should we support this or not?
+    # assert_equal(sk == "?", [False, False, False, False, True])
```

Comment:
Ah, yes, I guess that is not obvious, it is a bit like `equal_nan`... My tendency would be to start restrictive, i.e., if you want this you have to do `sk.symbol == "?"`.

Alternatively, I guess in `__eq__`, one could check whether `other` is of string type, and, if so, do the comparison with `.symbol`.

Reply:
I am ok with this not working and have documented comparing to `.symbol`.